app/core/Tracker2/SegmentTracker.js

```diff
@@ -103,49 +114,92 @@ export default class SegmentTracker extends BaseTracker {
   }
 
   async _initializeTracker () {
+    this.watchForDisableAllTrackingChanges(this.store)
+
     this.store.watch(
       (state, getters) => getters['me/isTeacher'],
-      this.onIsTeacherChanged.bind(this)
+      () => this.onIsTeacherChanged()
     )
 
     if (this.store.getters['me/isTeacher']) {
-      loadSegment()
+      this.onIsTeacherChanged(true)
       window.analytics.ready(this.onInitializeSuccess)
     } else {
       this.onInitializeSuccess()
     }
   }
 
-  async trackPageView (includeIntegrations = {}) {
+  async trackPageView (includeIntegrations = []) {
     await this.initializationComplete
 
-    if (!this.enabled) {
+    if (!this.enabled || this.disableAllTracking) {
       return
     }
 
+    const options = { ...DEFAULT_SEGMENT_OPTIONS }
+    this.addIntegrationsToSegmentOptions(options, includeIntegrations)
+
     const url = `/${Backbone.history.getFragment()}`
-    window.analytics.page(url, {}, {
-      integrations: {
-        All: false,
-        'Google Analytics': false
-      }
+    return new Promise((resolve) => {
+      window.analytics.page(undefined, url, {}, options, resolve)
     })
   }
 
+  // TODO legacy tracker filtered out identify calls when me.get('unsubscribedFromMarketingEmails') is true
   async identify (traits = {}) {
     await this.initializationComplete
 
-    if (!this.enabled) {
+    if (!this.enabled || this.disableAllTracking) {
       return
     }
+
+    const { me } = this.store.state
+
+    const {
+      _id,
+      ...meAttrs
+    } = me
+
+    const filteredMeAttributes = Object.keys(meAttrs)
+      .reduce((obj, key) => {
+        if (DEFAULT_SEGMENT_TRAITS_TO_REPORT.includes(key) && meAttrs[key] !== null) {
+          obj[key] = meAttrs[key]
+        }
+
+        return obj
+      }, {})
+
+    const options = { ...DEFAULT_SEGMENT_OPTIONS }
+    return new Promise((resolve) => {
+      window.analytics.identify(
+        this.store.state.me._id,
+        {
+          ...filteredMeAttributes,
+          ...traits
+        },
+        options,
+        resolve
+      )
+    })
   }
 
   async trackEvent (action, properties = {}, includeIntegrations = {}) {
     await this.initializationComplete
 
-    if (!this.enabled) {
+    if (!this.enabled || this.disableAllTracking) {
       return
     }
+
+    const options = { ...DEFAULT_SEGMENT_OPTIONS }
+    this.addIntegrationsToSegmentOptions(options, includeIntegrations)
```

Comment:
The `includeIntegrations` default value for the `trackEvent` function is an empty object literal here:
https://github.com/codecombat/codecombat/pull/5797/files#diff-9fb1ae5bb7b19f87f95e7c230fc75b0eR186

and has a default value of an array for the `addIntegrationsToSegmentOptions` of an array.

Could there be a type mismatch?

Reply:
Sure could!  Good catch.  Updated this and all of the other trackers.  I'd initially set the default to an empty object but that was wrong they should all be arrays.